scripts/Crawlers/FrdrCrawler.py

```diff
@@ -0,0 +1,532 @@
+        try:
+            self.is_completed(task['task_id'])
+        finally:
+            return destination_path
+
+    def _query_frdr(self):
+        """
+        Queries FRDR with a given request and obtains metadata
+        """
+        query = 'https://www.frdr-dfdr.ca/oai/request?verb=ListRecords&metadataPrefix=frdr'
+        results = requests.get(query)
+        # parse xml
+        results_dict = xmltodict.parse(results.text)
+        # load into json(dict) object
+        results_json = json.loads(json.dumps(results_dict))
+        if self.verbose:
+            logger.info("FRDR query: {}".format(query))
+        return results_json["OAI-PMH"]["ListRecords"]["record"]
+
+    def get_all_files_description(self, ep_id, ep_path):
+        """
+        Extracts information on file extensions and dataset size
+        :param ep_id: dataset globus endpoint
+        :param ep_path: dataset globus prefix path
+        """
+        files_types = []
+
+        def _get_contents(contents):
+            for content in contents:
+                if isinstance(content, dict):
+                    if "type" in content.keys() and content["type"] == "file":
+                        file_ext = str(content["name"].split(".")[1])
+                        if file_ext not in files_types:
+                            files_types.append(file_ext)
+
+                    elif "type" in content.keys() and content["type"] == "dir":
+                        _get_contents(content["contents"])
+                    else:
+                        continue
+                else:
+                    continue
+
+        # instantiate the file_size.json transfer
+        file_sizes_path = self.transfer_data(ep_id, ep_path, file_name='file_sizes.json')
+        # open file and read
+        with open(file_sizes_path) as json_file:
+            json_contents = json.load(json_file)
+        # access contents to get list of file types
+        _get_contents(json_contents["contents"])
+        dataset_size = json_contents["size"]
+        globus_path = json_contents["path"]
+        # remove file_sizes
+        os.remove(file_sizes_path)
+        return files_types, dataset_size, globus_path
+
+    def get_all_dataset_description(self):
+        """
+        Builds DATS for all dataset in the field of neuroscience
+        """
+        frdr_dois = []
+        datasets = self._query_frdr()
+        for dataset in datasets:
+            metadata = dataset["metadata"]["frdr:frdr"]
+
+            # TODO: to be updated: we are now testing 1 dataset
+            if "An End-to-end System for Automatic Characterization of Iba1 Immunopositive Microglia in Whole Slide Imaging: Data" in metadata["dc:title"]:
+
+                files_types, ds_size, globus_path = \
+                    self.get_all_files_description(metadata["frdr:globusEndpointName"],
+                                                   metadata["frdr:globusEndpointPath"])
+
+                frdr_dois.append(
```

Comment:
`ERROR    scripts.dats_validator.validator:validator.py:57 1 Validation error in distributions.0.access: 'landingPage' is a required property
`
Add a `landingPage` property.
See https://github.com/CONP-PCNO/conp-dataset/blob/master/scripts/Crawlers/OSFCrawler.py#L109